spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -907,12 +907,14 @@ def remove_trailing_spaces(self):
         cursor.endEditBlock()
 
     def fix_indentation(self):
-        """Replace tabs by spaces"""
+        """Replace tabs by 4 spaces."""

```

Comment:
@csabella isn't this fix dependent on the preference setting for what a tab should represent? Python is 4 but a user could change this to 3 or 2 if they follow google's ugly style guide.

Reply:
Yes, @goanpeca is right about this one.

Reply:
It doesn't use the settings now.  It's hard-coded as 4, which is why I made the comment.  It's in `sourcecode/utils.py`.
```
def fix_indentation(text):
    """Replace tabs by spaces"""
    return text.replace('\t', ' '*4)
```

Reply:
@csabella so then we need to fix fix_indentation perhpas?

Reply:
@goanpeca   Sure.  Should it be in this PR or a separate issue?

Reply:
I think @dalthviz fixed this one. @dalthviz?

Reply:
I fixed something regarding the use of preferences but in the comment/uncomment functionality in #5470, is that related with this?

Reply:
It's similar, but not the same.

Reply:
Ok, let's leave this for another PR then.

Reply:
Opened issue #5678 for the number of spaces.